server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/service/DeleteUsersDataOfDomainTask.java

```diff
@@ -0,0 +1,95 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Optional;
+
+import org.apache.james.core.Domain;
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskExecutionDetails;
+import org.apache.james.task.TaskType;
+import org.apache.james.user.api.UsersRepository;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class DeleteUsersDataOfDomainTask implements Task {
+    static final TaskType TYPE = TaskType.of("DeleteUsersDataOfDomainTask");
+
+    public static class AdditionalInformation implements TaskExecutionDetails.AdditionalInformation {
+        private final Instant timestamp;
+        private final Domain domain;
+
+        public AdditionalInformation(Instant timestamp, Domain domain) {
+            this.timestamp = timestamp;
+            this.domain = domain;
+        }
+
+        public Domain getDomain() {
+            return domain;
+        }
+
+        @Override
+        public Instant timestamp() {
+            return timestamp;
+        }
+    }
+
+    private final Domain domain;
+    private final DeleteUserDataService.Performer performer;
+    private final UsersRepository usersRepository;
+
+    public DeleteUsersDataOfDomainTask(DeleteUserDataService service, Domain domain, UsersRepository usersRepository) {
+        this.domain = domain;
+        this.performer = service.performer(Optional.empty());
+        this.usersRepository = usersRepository;
+    }
+
+    @Override
+    public Result run() {
+        return Flux.from(usersRepository.listReactive())
+            .filter(username -> username.getDomainPart()
```

Comment:
Ok with not a lot of domains with many users each ( ~ on prem ? )

Not OK for many domains with not a lot of users ( ~ b2c SaaS)

No need to solve this now but maybe we could add a default method into UsersRepository to list users of a domain...

So that later people running into this issue could use it and override this behaviour if needed.